service/frontend/workflowHandler.go

```diff
@@ -2176,27 +1827,47 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
+	rawHistoryQueryEnabled := wh.config.SendRawWorkflowHistoryByDomain(getRequest.GetDomain())
+
+	var history *gen.History
+	var historyBlob []*gen.DataBlob
 
-	history := &gen.History{}
-	history.Events = []*gen.HistoryEvent{}
 	if isCloseEventOnly {
 		if !isWorkflowRunning {
-			history, _, err = wh.getHistory(
-				scope,
-				domainID,
-				*execution,
-				lastFirstEventID,
-				nextEventID,
-				getRequest.GetMaximumPageSize(),
-				nil,
-				token.TransientDecision,
-				token.BranchToken,
-			)
-			if err != nil {
-				return nil, wh.error(err, scope)
+			if rawHistoryQueryEnabled {
+				historyBlob, _, err = wh.getRawHistory(
+					scope,
+					domainID,
+					*execution,
+					lastFirstEventID,
+					nextEventID,
+					getRequest.GetMaximumPageSize(),
+					nil,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+				if err != nil {
```

Comment:
This is for close event only, so you need to return only the last blob similar to line 1869.